Skill tree additions and restructurings (#50)
rust-lang/const-eval: src/skill_tree.md

name = "unconst_rules"

> label = "Need to come up\nwith a scheme\nfor doing unsafe\nin const fn"

Reviewer: ```suggestion
label = "Need to come up\nwith a scheme\nfor const unsafe/unconst"
```

items = [

>   { label = "transmute", href = "https://github.com/rust-lang/rust/issues/53605" },

Reviewer: ```suggestion
  { label = "transmute in functions", href = "https://github.com/rust-lang/rust/issues/53605" },
```

>   { label = "union field access", href = "https://github.com/rust-lang/rust/issues/51909", port = "union" },

Reviewer: ```suggestion
  { label = "union field access in functions", href = "https://github.com/rust-lang/rust/issues/51909", port = "union" },
```

  { label = "deref of raw pointers", href = "https://github.com/rust-lang/rust/issues/51911", port = "raw_ptr_deref" },
]
href = "https://github.com/rust-lang/const-eval/issues/14"

[[group]]
name = "offset_of"
label = "offset_of"
items = [
  { label = "offset_from", href = "https://github.com/rust-lang/rust/issues/41079", port = "offset_from" },
]
href = "https://github.com/rust-lang/rust/issues/71499"
requires = [
  "unconst_rules:raw_ptr_deref",
  "raw_ref_macros",

>   "maybe_uninit_as_ptr",

Reviewer: `const_offset_from` is missing.

label = "Mutex::new"
href = "https://github.com/rust-lang/rust/issues/66806"
items = []

> requires = ["parking_lot", "unconst_rules"]

Reviewer: Shouldn't `Mutex::new` still require `parking_lot`?
